pennylane/pulse/transmon.py

```diff
@@ -215,3 +217,285 @@ def __add__(self, other):
+
+    .. note:: Currently only supports ``d=2`` with qudit support planned in the future. For ``d=2`` we have :math:`a:=\frac{1}{2}(\sigma^x + i \sigma^y)`.
+
+    .. seealso::
+
+        :func:`~.drive`, :func:`~.rydberg_drive`, :func:`~.transmon_interaction`
+
+    Args:
+        amplitude (Union[float, callable]): The amplitude :math:`\Omega(t)`.
+            Can be a fixed number (``float``) or depend on time (``callable``)
+        phase (Union[float, callable]): The phase :math:`\phi(t)`.
+            Can be a fixed number (``float``) or depend on time (``callable``)
+        freq (Union[float, callable]): The drive frequency :math:`\nu`.
+            Can be a fixed number (``float``) or ``callable``. Physically it does not make sense for the drive frequency
+            to depend on time. The option for it to be ``callable`` is to allow the it to be a trainable parameter.
+        wires (Union[int, list[int]]): Label of the qubit that the drive acts upon. Can be a list of multiple wires.
+        d (int): Local Hilbert space dimension. Defaults to ``d=2`` and is currently the only supported value.
+
+    Returns:
+        :class:`~.ParametrizedHamiltonian`: a :class:`~.ParametrizedHamiltonian` representing the transmon interaction
+
+    **Example**
+
+    We can construct a drive term acting on qubit ``0`` in the following way. We parametrize the amplitude and phase
+    via :math:`\Omega(t) = \Omega \sin(\pi t)` and :math:`\phi(t) = \phi (t - \frac{1}{2})`.
+
+    .. code-block:: python3
+
+        def amp(Omega, t): return Omega * jnp.sin(jnp.pi*t)
+        def phase(phi, t): return phi * (t - 0.5)
+
+        H = qml.pulse.transmon_drive(amp, phase, 0, 0)
+
+        t = 0.5
+        Omega = 0.1
+        phi = 0.001
+        params = [Omega, phi]
+
+    Evaluated at :math:`t = \frac{1}{2}` with the parameters :math:`\Omega = 0.1` and :math:`\phi = 10^{-3}` we obtain
+    :math:`\Omega \left(\frac{1}{2}(\sigma^x + i \sigma^y) + \frac{1}{2}(\sigma^x + i \sigma^y)\right) = \Omega \sigma^x`.
+
+    >>> H(params, t)
+    (0.1*(PauliX(wires=[0]))) + (0.0*(-1*(PauliY(wires=[0]))))
+
+    We can combine ``transmon_drive()`` with :func:`~.transmon_interaction` to create a full driven transmon Hamiltonian.
+    Let us look at a chain of three transmon qubits that are coupled with their direct neighbors. We provide all numbers in
+    :math:`2\pi\text{GHz}`. We parametrize the amplitude as a sinusodial and make the maximum amplitude
+    as well as the drive frequency trainable parameters. We simulate the evolution for a time window of :math:`[0, 5]\text{ns}`.
+
+    .. code-block:: python3
+
+        omega = [5.1, 5., 5.3]
+        connections = [[0, 1], [1, 2]]
+        g = [0.02, 0.05]
+        H = qml.pulse.transmon_interaction(omega, connections, g, wires=range(3))
+
+        def amp(Omega, t): return Omega * jnp.sin(t)
+        def freq(fr, t): return fr
```

Comment:
This could be `qml.pulse.constant`, no? :)

Reply:
Could be yes, but I actually found this more self-explanatory